lib/utils/ssh.ts

```diff
@@ -36,17 +35,41 @@ export async function exec(
 	cmd: string,
 	stdout?: NodeJS.WritableStream,
 ): Promise<void> {
-	const command = `ssh \
-		-t \
-		-p 22222 \
-		-o LogLevel=ERROR \
-		-o StrictHostKeyChecking=no \
-		-o UserKnownHostsFile=/dev/null \
-		root@${deviceIp} \
-		${cmd}`;
+	const { which } = await import('./helpers');
+	const program = await which('ssh');
+	const args = [
+		'-n',
+		'-t',
+		'-p',
+		'22222',
+		'-o',
+		'LogLevel=ERROR',
+		'-o',
+		'StrictHostKeyChecking=no',
+		'-o',
+		'UserKnownHostsFile=/dev/null',
+		`root@${deviceIp}`,
+		cmd,
+	];
+	if (process.env.DEBUG) {
+		const logger = (await import('./logger')).getLogger();
+		logger.logDebug(`Executing [${program},${args}]`);
+	}
 
-	const stdio: StdioOptions = ['ignore', stdout ? 'pipe' : 'inherit', 'ignore'];
-	const { program, args } = getSubShellCommand(command);
+	// Note: stdin must be 'inherit' to workaround a bug in older versions of
+	// the built-in Windows 10 ssh client that otherwise prints the following
+	// to stderr and hangs: "GetConsoleMode on STD_INPUT_HANDLE failed with 6"
+	// They fixed the bug in newer versions of the ssh client:
+	// https://github.com/PowerShell/Win32-OpenSSH/issues/856
+	// but users whould have to manually download and install a new client.
+	// Note that "ssh -n" does not solve the problem, but should in theory
+	// prevent the ssh client from using the CLI process stdin, even if it
+	// is connected with 'inherit'.
+	const stdio: StdioOptions = [
+		'inherit',
+		stdout ? 'pipe' : 'inherit',
+		'inherit',
+	];
```

Comment:
Replacing 'ignore' with 'inherit' was the fix for the CLI hanging on Windows. Apparently it only happens because of a bug in the Microsoft ssh client that ships with Windows. To avoid ssh actually using the CLI process stdin, I've also added the `-n` flag when invoking `ssh`, although currently I don't think it matters.

I have also changed stderr from `ignore` to `inherit`. This was not necessary to avoid the hanging bug, but it means that error messages are not swept under the carpet. An error message regarding the `stdin` problem on Windows was actually being printed to stderr, but hidden. This made the investigation harder, as initially we had no idea that the problem was with the ssh client.